test/unit_test.py

```diff
@@ -147,6 +147,30 @@ def test_upstream_from_config(self):
         self.assertEqual('/project/g110/spack/upstream/daint_v0.18.1.5/base',
                          upstream_base)
 
+    def test_upstream_from_another_tag(self):
+        upstream_base = os.path.join(os.path.normpath(spack_c2sm_path),
+                                     'upstreams/daint/base')
+
+        self.assertEqual('/project/g110/spack/upstream/daint_v0.18.1.4/base',
+                         upstream_from_another_tag(upstream_base, 'v0.18.1.4'))
+
+        # mch_env_3 does not contain upstream_base -> None
+        self.assertEqual(None,
+                         upstream_from_another_tag(upstream_base, 'mch_env_3'))
+
+    def test_current_tag(self):
+        current_tag()
+
+    def test_git_version(self):
+        git_version()
+
+    def test_current_commit(self):
+        current_commit()
+
+    @unittest.skipUnless(2000 <= git_version(), 'needs git version > 2.0.0')
```

Comment:
git 3.0.0 wouldn't meet `2000 <= git_version()`.

Reply:
In this particular case, testing just the major version number would suffice.

Reply:
Good point. Changed it.